feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go

```diff
@@ -0,0 +1,586 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package setupservice is scoped only to be used for scripts in path
+// feature/security/gnsi/certz/tests/client_certificates
+// Do not use elsewhere.
+package setupservice
+
+import (
+	context "context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+	spb "github.com/openconfig/gnoi/system"
+	authzpb "github.com/openconfig/gnsi/authz"
+	certzpb "github.com/openconfig/gnsi/certz"
+	gribipb "github.com/openconfig/gribi/v1/proto/service"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/knebind/creds"
+	p4rtpb "github.com/p4lang/p4runtime/go/p4/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	username = "certzuser"
+	password = "certzpasswd"
+	sn       = "role001.pop55.net.example.com"
+	servers  []string
+	retries  int
+)
+
+type rpcCredentials struct {
+	*creds.UserPass
+}
+
+func (r *rpcCredentials) GetRequestMetadata(_ context.Context, _ ...string) (map[string]string, error) {
```

Comment:
it's been a while and I may have asked this previously, but why does this function need to accept a context.Context which it never uses?

Is this mocking some production version of this function?
This looks like a whole new package, and the tests that appear to use this just use this directly by name, why add to the function signature something that just isn't used?

Reply:
no reply here, what's happening with this?